src/controller/CHIPCommissionableNodeController.cpp

```diff
@@ -35,5 +37,39 @@ const Mdns::DiscoveredNodeData * CommissionableNodeController::GetDiscoveredComm
     return GetDiscoveredNode(idx);
 }
 
+CHIP_ERROR CommissionableNodeController::RequestCommissioning(int commissionerNumber)
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY_CLIENT
+    const Mdns::DiscoveredNodeData * commissioner;
+    for (int i = 0; i < CHIP_DEVICE_CONFIG_MAX_DISCOVERED_NODES; i++)
+    {
+        commissioner = GetDiscoveredCommissioner(i);
+        if (commissioner != nullptr && --commissionerNumber == 0)
+        {
+            break;
+        }
+    }
+    if (commissionerNumber != 0)
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    // Enter commissioning mode, open commissioning window
+    InitServer();
+
+    // Send User Directed commissioning request
+    chip::Transport::PeerAddress commissionerAddress =
+        chip::Transport::PeerAddress::UDP(commissioner->ipAddress[0], commissioner->port);
```

Comment:
This needs an interface id too, if we discovered a link-local ipv6 address.  `PacketDataReporter` should be storing that in the `DiscoveredNodeData` like it does in the `ResolvedNodeData`, no?

Please make sure to test this scenario.

Reply:
Stored the interfaceIds